src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,337 @@
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import static edu.harvard.iq.dataverse.util.json.NullSafeJsonBuilder.jsonObjectBuilder;
+
+public class SignpostingResources {
+    private static final Logger logger = Logger.getLogger(SignpostingResources.class.getCanonicalName());
+    SystemConfig systemConfig;
+    DatasetVersion workingDatasetVersion;
+//    JsonObject licJsonObj;
+    JsonObject describedByJsonObj;
+    Boolean useDefaultFileType;
+    String defaultFileTypeValue;
+    int maxAuthors;
+    int maxItems;
+
+    public SignpostingResources(SystemConfig systemConfig, DatasetVersion workingDatasetVersion, String jsonSetting) {
+        this.systemConfig = systemConfig;
+        this.workingDatasetVersion = workingDatasetVersion;
+        if (jsonSetting == null) {
+            jsonSetting = BundleUtil.getStringFromBundle("signposting.configuration.SignpostingConf");
+        }
+        JsonReader jsonReader = Json.createReader(new StringReader(jsonSetting));
+        JsonObject spJsonSetting = jsonReader.readObject();
+        jsonReader.close();
+//        licJsonObj = spJsonSetting.getJsonObject("license");
+        describedByJsonObj = spJsonSetting.getJsonObject("describedby");
+        useDefaultFileType = spJsonSetting.getBoolean("useDefaultFileType", true);
+        defaultFileTypeValue = spJsonSetting.getString("defaultFileTypeValue", "https://schema.org/Dataset");
+        maxAuthors = spJsonSetting.getInt("maxAuthors", 5);
+        maxItems = spJsonSetting.getInt("maxItems", 5);
+    }
+
+    /**
+     * Get identifier schema for each author
+     * <p>
+     * For example:
+     * if author has VIAF
+     * Link: <http://viaf.org/viaf/:id/>; rel="author"
+     *
+     * @param datasetAuthors list of all DatasetAuthor object
+     * @return all the non empty author links in a string
+     */
+    private String getAuthors(List<DatasetAuthor> datasetAuthors) {
+        String singleAuthorString;
+        String identifierSchema = "";
+        int visibleAuthorCounter = 0;
+//        if (datasetAuthors.size() > maxAuthors) {return "";}
+        for (DatasetAuthor da : datasetAuthors) {
+            logger.info(String.format(
+                    "idtype: %s; idvalue: %s, affiliation: %s; identifierUrl: %s",
+                    da.getIdType(),
+                    da.getIdValue(),
+                    da.getAffiliation(),
+                    da.getIdentifierAsUrl()
+            ));
+
+            String authorURL = "";
+            authorURL = getAuthorUrl(da);
+            if (!Objects.equals(authorURL, "")) {
+                visibleAuthorCounter++;
+                // return empty if number of visible author more than max allowed
+                if (visibleAuthorCounter >= maxAuthors) return "";
```

Comment:
So the intent is to show no authors if there are more than max and not the first max authors? (And we're only talking about authors with URIs? If there are ten authors, three who have ORCIDs, you'd still show 3 authors for a max of 5?)

If so, reusing getJsonAuthors might involve adding a param to that method to check the max (if I understand, the level1 header uses the max limit but the linkset always shows however many there are?)

Reply:
same as above. in level 2, the full json structure is needed. In level 1, it's string format, not json. 
Only authors with links should be counted and displayed in either level 1 or 2.